torch_xla/csrc/init_python_bindings.cpp

```diff
@@ -3336,19 +3368,17 @@ void InitXlaModuleBindings(py::module m) {
                     opt_device ? &opt_device.value() : nullptr);
             return check_materialization_helper(xtensors);
           })
-      .def(
-          "_get_graph_hash",
-          [](const std::vector<at::Tensor>& tensors) {
-            std::vector<XLATensorPtr> xtensors;
-            xtensors.reserve(tensors.size());
-            for (auto& tensor : tensors) {
-              xtensors.push_back(GetValueOrThrow(bridge::GetXlaTensor(tensor)));
-            }
-            torch::lazy::hash_t hash =
-                XLAGraphExecutor::Get()->GetGraphHash(xtensors);
-            std::string bin((const char*)&hash, sizeof(hash));
-            return py::bytes(bin);
-          })
+      .def("_get_graph_hash",
+           [](const std::vector<at::Tensor>& tensors)
+               -> absl::StatusOr<py::bytes> {
```

Comment:
I'm not sure about the motivation for allowing the callback to return a StatusOr. Since the callback is invoked by pytorch from python, there is no benefit for returning a StatusOr compared with TORCH_CHECK(). The biggest benefit seems to be that it lets us use XLA_ASSIGN_OR_RETURN in the callback, but we can use MaybeThrow instead.

Reply:
I would argue that there are 2 main benefits:

1. As you mentioned, we get to call `XLA_ASSIGN_OR_RETURN()`
2. We get to overwrite the error message, and see the source location when `TORCH_SHOW_CPP_STACKTRACES` is set

While we could modify `MaybeThrow()` for doing (1), for (2) we would need to introduce another macro. Therefore, I think that calling `XLA_ASSIGN_OR_RETURN()` is better.

Reply:
I still don't see the benefits:

- (1) is not a benefit as we can use MaybeThrow().
- for (2) we can just call TORCH_CHECK() directly.